test/e2e/shoot/internal/logging/event_logging.go

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/gardener/gardener/test/framework"
+	"github.com/go-logr/logr"
+
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	errorsutil "github.com/onsi/gomega/gstruct/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	eventMessageForSeed  = "Event from Event Logger integration test related to the Seed"
+	eventMessageForShoot = "Event from Event Logger integration test related to the Shoot"
+)
+
+// EventLoggingVerifier verifies the event-logger.
+type EventLoggingVerifier struct {
+	*framework.ShootFramework
+	shootNamespace string
+	seedEvent      *corev1.Event
+	shootEvent     *corev1.Event
+}
+
+// Verify verifies that the event logging is working properly
+func (v *EventLoggingVerifier) Verify(ctx context.Context) {
+	v.before(ctx)
+	v.prepare(ctx)
+	v.expect(ctx)
+}
+
+// After cleans all resources created in prepare function
+func (v *EventLoggingVerifier) After(ctx context.Context) {
+	By("Delete seed and shoot events")
+	Eventually(func(g Gomega) {
+		g.Expect(v.ShootFramework.SeedClient.Client().Delete(ctx, v.seedEvent)).To(Or(Succeed(), BeNotFoundError()))
+		g.Expect(v.ShootFramework.ShootClient.Client().Delete(ctx, v.shootEvent)).To(Or(Succeed(), BeNotFoundError()))
```

Comment:
The shoot might already be deleted when this is executed.
Please make sure it will succeed in this case as well.

Reply:
How?